service/csv/csv.go

```diff
@@ -0,0 +1,93 @@
+package csv
+
+import (
+	"encoding/csv"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/rodrigomkd/go-rest-api/model"
+)
+
+type ICSVService interface {
+	ReadCSV(path string) ([][]string, error)
+	SaveCSV(path string, records [][]string)
+	SaveActivities(path string, activities []model.Activity)
+}
+
+type CSVService struct {
+	csv ICSVService
+}
+
+func New() *CSVService {
+	return &CSVService{}
+}
+
+//Read CSV file
+func (cs CSVService) ReadCSV(path string) ([][]string, error) {
+	csvFile, err := os.Open(path)
+	if err != nil {
+		log.Print("Error to open CSV file: ", err)
+	}
+
+	log.Print("Successfully Opened CSV file")
+	defer csvFile.Close()
+
+	return csv.NewReader(csvFile).ReadAll()
+}
+
+func (cs CSVService) SaveCSV(path string, records [][]string) {
+	if _, err := os.Stat(path); os.IsExist(err) {
+		os.Remove(path)
+	}
+
+	file, err := os.Create(path)
+	checkError("Cannot create file: ", err)
+	defer file.Close()
+
+	writer := csv.NewWriter(file)
+	defer writer.Flush()
+
+	err = writer.WriteAll(records)
+	checkError("Cannot write to file", err)
+}
+
+func (cs CSVService) SaveActivities(path string, activities []model.Activity) {
+	if _, err := os.Stat(path); os.IsExist(err) {
+		os.Remove(path)
+	}
+
+	file, err := os.Create(path)
```

Comment:
Please do a correct handling of the errors, if something wrong happened then you would like to stop the flow and return the error as well

Reply:
This applies to pretty much all of the file